main.go

```diff
@@ -156,45 +156,48 @@ func main() {
 	}
 
 	//+kubebuilder:scaffold:builder
-
-	// Create OCSInitialization CR if it's not present
-	client := mgr.GetClient()
-	releaseDscInitialization := &dsci.DSCInitialization{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "DSCInitialization",
-			APIVersion: "dscinitialization.opendatahub.io/v1alpha1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "default",
-		},
-		Spec: dsci.DSCInitializationSpec{
-			ApplicationsNamespace: dscApplicationsNamespace,
-			Monitoring: dsci.Monitoring{
-				Enabled: false,
+	// Check if Auto configuration is enabled
+	if os.Getenv("ENABLE_AUTO_CONFIG") == "" || os.Getenv("ENABLE_AUTO_CONFIG") != "false" {
```

Comment:
I wonder if it wouldn't be better to check == "true" for proceeding, == "false" for disabling and failing if it is not true or false? it would make intent explicit and avoid silent failures on this critical flag?

Reply:
When a user installs the operator through OperatorHub, subscription is created for the user by default. That will not include this config. In that case, if we set up failure for not defining the flag, it will result in operator install failure

Reply:
Sorry, what I mean is if the user makes a typo like: ENABLE_AUTO_CONFIG=flse . With the current code, although the user intended to disable auto-config, it will silently do it. 
I completely agree with you that in the absence of the flag, it should use the default behaviour of installing everything.

Reply:
Updated